lib/Util.php

```diff
@@ -65,31 +65,27 @@ public static function getTurnSettings(IConfig $config, $uid) {
 	 * Generates a username and password for the TURN server based on the
 	 *
 	 * @param IConfig $config
-	 * @param ISession $session
 	 * @param ITimeFactory $timeFactory
 	 * @return array
 	 */
-    public static function generateTurnSettings(IConfig $config, ISession $session, ITimeFactory $timeFactory) {
+    public static function generateTurnSettings(IConfig $config, ITimeFactory $timeFactory) {
 		// generate from shared secret
 		$turnServer = $config->getAppValue('spreed', 'turn_server', '');
 		$turnServerSecret = $config->getAppValue('spreed', 'turn_server_secret', '');
 		$turnServerProtocols = $config->getAppValue('spreed', 'turn_server_protocols', '');
-		$sessionId = $session->get('spreed-session');
 
-		if ($turnServer === '' || $turnServerSecret === '' || $turnServerProtocols === '' || empty($session)) {
+		if ($turnServer === '' || $turnServerSecret === '' || $turnServerProtocols === '') {
 			return array();
 		}
 
-		$time = $timeFactory->getTime();
-
 		// the credentials are valid for 24h - FIXME add the TTL to the response and properly reconnect then
-		$string = sprintf('%d:%s', $time + 86400, $sessionId);
-		$hashedString = hash_hmac('sha1', $string, $turnServerSecret, true);
+		$username = $timeFactory->getTime() + 86400;
```

Comment:
Also please consider bringing back a unique id per user so the TURN server can enforce limits per real user instead of only per server. How this is implemented also has implications for data privacy and be required in some situations.